### ISSUE TITLE
Breakpoints that are set before program run cannot be deleted/disabled.

### ISSUE THREAD

**Issue by aonurK** [microsoft/vscode-cpptools#9147]
Bug type: Debugger

Does this issue occur when all extensions are disabled?: Yes (Only the C/C++ extension is enabled and issue occured)

VS Code Version: 1.66
OS Version: Ubuntu 20.04.4 LTS
GDB version: 9.2
C/C++ Extension version: 1.9.7
When I run the program without debugging it works as expected. Also, when I set the breakpoints during application run it hits the breakpoint and after I removed the breakpoints it does not hit again. So far so good,and here are the steps to create this issue.

1) Set the breakpoint before application run
2) Debug the program,
3) It hits the breakpoint
4) Remove the breakpoint,
5) It hits the breakpoint again!

In the breakpoints menu, as it is shown, nothing can be seen. But, when I opened the debug console and write the "-exec info break" to see current breakpoints, it shows the deleted breakpoints that is set before application run. When I manually remove it with (lets say breakpoint id is 2) "-exec d 2" it is deleted and program continues as expected.

To sum up, it seems like there exists unhealthy communication between vs code and gdb. The below figures better explains what I mean.

I investigated through internet but did not find a solution for that except for manually deleting which is time consuming and annoying workaround for me.
![1](https://user-images.githubusercontent.com/16953239/161926656-d0dba0d0-b22a-448d-9a66-b6ba5b2942ed.png)
![2](https://user-images.githubusercontent.com/16953239/161926666-bfa5e998-2557-45c7-8006-11ce2d5d3c5f.png)

**Comment by botanyldrm** [comment]
I have same configıuration and everything works like a charm!

**Comment by WardenGnaw** [comment]
Duplicate of https://github.com/microsoft/vscode-cpptools/issues/9095

The current workaround is downgrading to 1.7.1

**Comment by lokimon** [comment]
Seeing the exact same thing.
Set some breakpoints
```
-exec info breakpoints
Num     Type           Disp Enb Address            What
2       breakpoint     keep y   <PENDING>          ds_common.h:2728
3       breakpoint     keep y   0x00007ffff7505bb0 in targetd_msg_callback at ../src/targetd.c:202
	breakpoint already hit 1 time
4       breakpoint     keep y   0x00007ffff7505c3c in targetd_msg_callback at ../src/targetd.c:211
5       breakpoint     keep y   0x00007ffff7505e73 in targetd_msg_callback at ../src/targetd.c:232

```
Then if I remove breakpoint on line 202 I get
```
-exec info breakpoints
Num     Type           Disp Enb Address            What
2       breakpoint     keep y   <PENDING>          ds_common.h:2728
3       breakpoint     keep y   0x00007ffff7505bb0 in targetd_msg_callback at ../src/storage/storage_targetd.c:202
	breakpoint already hit 1 time
4       breakpoint     keep y   0x00007ffff7505c3c in targetd_msg_callback at ../src/storage/storage_targetd.c:211
5       breakpoint     keep y   0x00007ffff7505e73 in targetd_msg_callback at ../src/storage/storage_targetd.c:232
6       breakpoint     keep y   0x00007ffff7505c3c in targetd_msg_callback at ../src/storage/storage_targetd.c:211
7       breakpoint     keep y   0x00007ffff7505e73 in targetd_msg_callback at ../src/storage/storage_targetd.c:232
```

Seems like it doesn't remove a breakpoint, but instead sends the list of current breakpoints again.
